scipy/sparse/linalg/matfuncs.py

```diff
@@ -592,6 +592,18 @@ def _expm(A, use_exact_onenorm):
     if len(A.shape) != 2 or A.shape[0] != A.shape[1]:
         raise ValueError('expected a square matrix')
 
+    # Trivial case
+    if A.shape == (1, 1):
+        data = [[np.exp(A[0, 0])]]
+
+        # 'ndarray' constructor does not accept
+        # a 'data' argument, so we need to call
+        # 'np.array' instead
+        if type(A).__name__ == 'ndarray':
```

Comment:
Let's do it like this instead:
```
if A.shape == (1, 1):
    res = np.exp([[A[0,0]]])
    if isspmatrix(A):
        return A.__class__(res)
    else:
        return res
```
The other code path does not preserve ndarray subclasses either, so best to not worry about subclasses for now.
They would require special handling, cf., https://github.com/numpy/numpy/blob/master/numpy/core/fromnumeric.py#L41